c7n/filters/related.py

```diff
@@ -86,16 +94,20 @@ def process_resource(self, resource, related):
             if self.match(robj):
                 found.append(rid)
 
-        if self.AnnotationKey is not None and found:
-            akey = 'c7n:%s' % self.AnnotationKey
-            resource[akey] = list(set(found).union(resource.get(akey, [])))
+        if found:
+            self.add_annotations(found, resource)
 
         if op == 'or' and found:
             return True
         elif op == 'and' and len(found) == len(related_ids):
             return True
         return False
 
+    def add_annotations(self, related_ids, resource):
```

Comment:
can you make this method _private (leading underscore), annotation format/schema is filter/action specific.